plugins/outputs/http/http_test.go

```diff
@@ -526,6 +527,89 @@ func TestOAuthClientCredentialsGrant(t *testing.T) {
 	}
 }
 
+func TestOAuthAuthorizationCodeGrant(t *testing.T) {
+	ts := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.Path {
+		case "/":
+			w.WriteHeader(http.StatusAccepted)
+		case "/token":
+			w.WriteHeader(http.StatusAccepted)
+		default:
+			w.WriteHeader(http.StatusNotFound)
+		}
+	}))
+	defer ts.Close()
+
+	l, err := net.Listen("tcp", "localhost:58025")
```

Comment:
Is it required to set the port here? The reason I'm asking is that if two tests run in parallel one might fail due to the port being already taken...

Reply:
The original reason I set a static port is because the `token_uri` in the key file is used to request the access token. Since I couldn't update the port of the token_uri inside the key file, I pinned it. 

I have since refactored this test to use `func (c *common) TempDir() string` and `func CreateTemp(dir string, pattern string) (*File, error)`, which obviates the need for the key file in the testdata folder.